source/contributing/implementation/debugging.rst

```diff
@@ -4,6 +4,39 @@ Debugging Sponge Within the IDE
 
 .. note::
 
-    This page has not been filled yet by our Documentation Team. If you feel like you can help, you can do so on `our
-    GitHub repository <https://github.com/spongepowered/spongedocs>`_. Also see the `related GitHub Issue
+    This documentation is not yet complete see the `related GitHub Issue
     <https://github.com/SpongePowered/SpongeDocs/issues/356>`_ for more information on what is required.
+
+Debugging SpongeCommon
+======================
+
+By the point you debug, you should already finish making changes in your own branch in your fork (see
+:doc:`../howtogit`).
+
+Unfortunately, SpongeCommon exists as an abstract layer to reduce duplicate code between SpongeForge and SpongeVanilla;
+thus, it is not for production environments, and you cannot run SpongeCommon directly. However, you can run SpongeForge
+or SpongeVanilla with the changes you've applied in SpongeCommon. Below is a description on testing SpongeCommon changes
+on SpongeForge.
+
+Assume you have a branch ``fix/foo``, based on ``bleeding`` branch of ``SpongePowered/SpongeCommon``, on your
+SpongeCommon fork. You want to test it against the corresponding SpongeForge (or the ``bleeding`` branch of it). The
+procedure is as follows:
+
+1. Clone SpongeForge repository from GitHub: Find a clean workspace, run 
```

Comment:
I wonder if RST will interpret this as three lists and start each one at `1.`. We shall see...

Reply:
I recall @Spongy builds prs automatically. Doesn't it?

Reply:
Hmm. appears spongy doesn't like me (no branch `661`)